src/zernike_polynomial.cpp

```diff
@@ -1,12 +1,30 @@
 #include <tallies/zernike_polynomial.hpp>
+#include <utils/constants.hpp>
 
 ZernikePolynomials::ZernikePolynomials(std::size_t order)
     : order_(order), max_n_(), Zr_coefficients_(), m_(), n_(), Zr_types_() {
   // Evaluate the polynomial coefficients up to given order
   // Zr_coefficients_ will be having the coeff. stating from lowest power to
   // higher for each order, it will be useful this way as evaluating the power
   // can easily be done.
-  Zr_coefficients_.reserve(order_ + 1);
+
+  // get a maximum size to store the coefficients
+  std::pair<int, int> high_n_and_l = get_n_and_l(order_);
+  // store the n for highest order
+  // this is required as we have to evaluate upto this power
+  max_n_ = static_cast<std::size_t>(high_n_and_l.first);
```

Comment:
Why do we immediately cast these ints to an std::size_t ? Can the method get_n_and_l immediately return a pair of std::size_t ?